[21990,22000] Custom fields errors in split pane

oliverguenther opened the pull request:
This PR addresses two issues:
1. List, Version and User properties were not set as `writable` since https://github.com/opf/openproject/pull/3728. They were simply missed there and I 'fixed' the spec incorrectly, thus effectively hiding the missing property.
2. The drop-down has been altered to use embedded values where applicable in order to react to sortable lists (e.g., version, which have a `definingProject` property for grouping). This PR extends this directive to respect the values of custom fields as correct item names.

This should fix both
https://community.openproject.org/work_packages/21990
https://community.openproject.org/work_packages/22000

oliverguenther committed:
Add writable to other custom field injectors

This was missed in https://github.com/opf/openproject/pull/3728
and properly disguised by a faulty spec fix from my side.

oliverguenther committed:
Properly show embedded values in changed drop-down directive

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://oliverguenther-fix-21990-custom-fields.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://oliverguenther-fix-21990-custom-fields.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

basiszwo commented:
Tested and it is not fully working:
- [x] I created the required custom fields in the demo project (all as required fields and active for all projects)
- [x] type user does not show project members; select field is empty
- [x] type list: only works when a default is set
- [x] type version: the demo project already contains versions but they do not show up
- [x] editing the fields throws the error too

Getting the following JS error

```
TypeError: Cannot read property 'props' of null
    at InplaceEditorDropDownController.addHrefTracker
```

oliverguenther committed:
Avoid nRepeat dupes error

Solve `[ngRepeat:dupes] Duplicates in a repeater are not allowed.`
by adding a `track by $index`.

oliverguenther committed:
Append empty option appropriately

Even when the field the drop-down is used with is required by the
schema, we need to have a trackable empty option that will be set
initially.
